netsim-ffi/src/lib.rs

```diff
@@ -20,7 +19,9 @@ pub enum SimError {
 
     /// The function is not yet implemented, please report this issue
     /// to maintainers
-    NotImplemented,
+    NotImplemented = 4,
+
+    NoMoreData = 5
```

Comment:
I don't think this is the right _name_ for this error code. This is used every time the Context has:

1. stopped: by design or error. if the Context has been shutdown but the Sockets are still in use, the `send` or `receive` will fail
2. the internal mutexes may have been poisoned (a panic happened while a mutex was locked).

`1.` is likely to happen often: I can see users starting simulations and stopping the Context (calling the shutdown function) and after that shutting down the "nodes". I think `SocketDisconnected` might be more descriptive of the underlying issue. what do you think ?